backend/app/core/authz/authorization.py

```diff
@@ -20,7 +23,53 @@
 
 from .actions import AuthorizationAction
 
-Model: TypeAlias = Union[Type[DataProduct], Type[Dataset], None]
+Model: TypeAlias = Union[Type[DataProduct], Type[Dataset], Type[DataOutput], None]
+
+
+class SubjectResolver(ABC):
+    DEFAULT: str = "*"
+
+    @abstractmethod
```

Comment:
This can/should be decorated with `@classmethod` as well.
Reference: https://docs.python.org/3/library/abc.html#abc.abstractmethod

```suggestion
    @classmethod
    @abstractmethod
```

Reply:
nice, I tried it the other way around and that did not work. Was not smart enough to turn the order around :P